src/WebJobs.Extensions.DurableTask.CodeGeneration/README.md

```diff
@@ -0,0 +1,215 @@
+# CodeGeneration for Durable Functions
+
+Assists in performing Roslyn-based code generation for apps using the [Durable Function](https://github.com/Azure/azure-functions-durable-extension) extension for the [Durable Task](https://github.com/Azure/durabletask) framework. Automatically keeps dependencies between orchestration/actvity functions up to date as your code evolves.
+
+Supports design-time code generation that responds to changes made in hand-authored files by generating new code that shows up to Intellisense as soon as you type.
+
+## What It Does
+
+Automatically generates method stubs that correspond to the contracts of Orchestration/Activity functions it finds within the project. Ensures that the return type, functionName, and parameters all stay up to date as your project grows.
+
+## How to use
+
+1. Add the project's nuget package as a dependency. Mark the reference with `OutputItemType="Analyzer"` and `ReferenceOutputAssembly="false"`.
+
+For example: 
+
+```xml
+<ProjectReference Include="..\WebJobs.Extensions.DurableTask.CodeGen.SourceGenerator\WebJobs.Extensions.DurableTask.CodeGen.SourceGenerator.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+```
+
+*The project should now automatically generate code for Orchestration/Activity functions.*
+
+2. *Optionally*, add the following statement to a property group in your project's csproj in order to see the generated files.
+
+```xml
+<CompilerGeneratedFilesOutputPath>$(BaseIntermediateOutputPath)Generated</CompilerGeneratedFilesOutputPath>
+```
+
+3. Add an Orchestration/Activity function to your project.
+
+```csharp
+[FunctionName("SayHello")]
+public static string SayHello([ActivityTrigger] IDurableActivityContext context, ILogger log)
+{
+    var name = context.GetInput<string>();
+
+    log.LogInformation($"Saying hello to {name}.");
+    return $"Hello {name}!";
+}
+```
+
+4. Manually add using statement to reference the generated code.
+
+```
+using Microsoft.Azure.Webjobs.Extensions.DurableTask.Generated;
+```
+
+5. Replace ```IDurableClient``` / ```IDurableOrchestrationContext``` usage for their generated counterparts ```IGeneratedDurableClient``` / ```IGeneratedDurableOrchestrationContext```. The generated interfaces can perform all operations exposed by standard interfaces, in additional to performing typed calls to Orchestration/Activity Function.
+
+Some example Activity calls through the generated interface:
+
+```csharp
+[FunctionName("SimpleOrchestration")]
+public static async Task<List<string>> SimpleOrchestrator(
+    [OrchestrationTrigger] IGeneratedDurableOrchestrationContext context)
+{
+    var outputs = new List<string>();
+
+    // Replace "hello" with the name of your Durable Activity Function.
+    outputs.Add(await context.Activity.SayHello("Tokyo"));
```

Comment:
Sorry if this has already been discussed, but would it make sense to pluralize the name of this property, e.g. `Activities` instead of `Activity`? That feels a bit more natural to me.

Reply:
Not discussed. Easy change. Do you want ```Orchestrations``` too?

Reply:
Yes, that would be great!